bundles/AdminBundle/Resources/public/js/pimcore/element/helpers/gridConfigDialog.js

```diff
@@ -110,18 +110,29 @@ pimcore.element.helpers.gridConfigDialog = Class.create({
             });
         }
 
+        var windowWidth = 950;
         this.window = new Ext.Window({
-            width: 950,
+            width: windowWidth,
+            maxWidth: Ext.getBody().getViewSize().width,
             height: '95%',
+            maxHeight: Ext.getBody().getViewSize().height,
             modal: true,
             title: t('grid_options'),
             layout: "fit",
             items: [this.tabPanel],
-            buttons: buttons
+            buttons: buttons,
+            constrainHeader: true
         });
 
         this.window.show();
         this.updatePreview();
+
+        window.addEventListener('resize', () => {
+            this.window.setWidth(windowWidth);
+            this.window.setMaxWidth(Ext.getBody().getViewSize().width);
+            this.window.setMaxHeight(Ext.getBody().getViewSize().height);
+            this.window.center();
+        });
```

Comment:
Maybe my JS noobiness but why do we have to set the same values on the `window` here that are set above (line 113)?

Reply:
if you are referring to lines 132-133 it is because it is checking the window size every frame of the browser resize animation and adding it to the window object.

Reply:
Got it - this is the browser resize event which you're listening to in order to update the max size of the popup